bitarray/bitmap.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright (c) 2016, Theodore Butler
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+* Redistributions of source code must retain the above copyright notice, this
+  list of conditions and the following disclaimer.
+
+* Redistributions in binary form must reproduce the above copyright notice,
+  this list of conditions and the following disclaimer in the documentation
+  and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+// Package bitmap contains bitmaps of length 32 and 64 for tracking bool
+// values without the need for arrays or hashing.
+package bitarray
+
+// Bitmap32 tracks 32 bool values within a uint32
+type Bitmap32 uint32
+
+// SetBit returns a Bitmap32 with the bit at the given position set to 1
+func (b Bitmap32) SetBit(pos uint) Bitmap32 {
+	return b | (1 << pos)
+}
+
+// ClearBit returns a Bitmap32 with the bit at the given position set to 0
+func (b Bitmap32) ClearBit(pos uint) Bitmap32 {
+	return b & ^(1 << pos)
+}
+
+// HasBit returns true if the bit at the given position in the Bitmap32 is 1
+func (b Bitmap32) HasBit(pos uint) bool {
```

Comment:
We use the name `GetBit` elsewhere in the code to be consistent with our convention of `SetBit`. Mind changing it here as well?